specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json

```diff
@@ -0,0 +1,4191 @@
+          "readOnly": true
+        }
+      }
+    },
+    "ClusterList": {
+      "description": "A paged list of clusters",
+      "properties": {
+        "value": {
+          "description": "The items on a page",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/Cluster"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next page if any",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "AdminCredentials": {
+      "description": "Administrative credentials for accessing vCenter and NSX-T",
+      "properties": {
+        "nsxtUsername": {
+          "description": "NSX-T Manager username",
+          "type": "string",
+          "readOnly": true
+        },
+        "nsxtPassword": {
+          "description": "NSX-T Manager password",
+          "type": "string",
+          "readOnly": true,
+          "x-ms-secret": true
+        },
+        "vcenterUsername": {
+          "description": "vCenter admin username",
+          "type": "string",
+          "readOnly": true
+        },
+        "vcenterPassword": {
+          "description": "vCenter admin password",
+          "type": "string",
+          "readOnly": true,
+          "x-ms-secret": true
+        }
+      }
+    },
+    "Sku": {
+      "description": "The resource model definition representing SKU",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU."
+        }
+      },
+      "required": [
+        "name"
+      ]
+    },
+    "StoragePolicy": {
+      "description": "The storage policy for the cluster",
+      "type": "string"
+    },
+    "StoragePolicyList": {
+      "description": "A list of available storage policies",
+      "properties": {
+        "value": {
+          "description": "The total storage policies",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/StoragePolicy"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next page if any",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "HcxEnterpriseSiteList": {
+      "description": "A paged list of HCX Enterprise Sites",
+      "properties": {
+        "value": {
+          "description": "The items on a page",
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/HcxEnterpriseSite"
+          }
+        },
+        "nextLink": {
+          "description": "URL to get the next page if any",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "HcxEnterpriseSite": {
+      "description": "An HCX Enterprise Site resource",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
@pilor, does this need to be marked as `readOnly` since all its properties are `readOnly`. I'm getting a [failure from LiveValidation](https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8692384). cc @leedouglas

Reply:
The PUT for HcxEnterpriseSite is marked as a generic object without any definition... that should be fixed and should fix the validation failure as well

Reply:
Okay, I committed d6be8ad which is a different solution to PUT an empty body of `{}`. I turned off flattening, since all the properties are readOnly and the combo does not work with SDKs like Python: https://github.com/Azure/autorest.python/issues/653

Reply:
I'm going to try this out with Python and C#, before I port it back to the 2020-03-20 API.

Reply:
Python code generated with `--use=@microsoft.azure/autorest.python@3.0.62` still has a problem. Code generated with `--use=@autorest/python@5.3.0` did not, which is what the the Python SDK uses. Users of the older autorest.python like the az-vmware-cli extension will just need to upgrade.